optuna/visualization/_contour.py

```diff
@@ -113,6 +121,12 @@ def _get_contour_plot(study: Study, params: Optional[List[str]] = None) -> "go.F
         figure = go.Figure(data=sub_plots, layout=layout)
         figure.update_xaxes(title_text=x_param, range=param_values_range[x_param])
         figure.update_yaxes(title_text=y_param, range=param_values_range[y_param])
+
+        if _is_categorical(trials, x_param) and update_category_axes[x_param]:
```

Comment:
I think that the following line can be simplified (and optimized) as follows:
```suggestion
        if update_category_axes.get(x_param, False):
```